dwave/cloud/api/client.py

```diff
@@ -312,11 +312,11 @@ def __init__(self, **config):
         self.session = self._create_session(self.config)
 
     @classmethod
-    def from_client_config(cls, config: ClientConfig, **kwargs):
-        """Create class instance based on
+    def from_config_model(cls, config: ClientConfig, **kwargs):
```

Comment:
I wonder if these `from_config_X` methods could all be replaced by `.from_config()` and `functools.singledispatchmethod`. I know we do a very similar pattern in other parts of Ocean but I can't say that I've ever found it to be aesthetically satisfying.

There may be other considerations I am missing of course.

Reply:
I tried exactly that when splitting config into these submethods (which should see very rare use) in #568, but couldn't get the no-argument option (default `config=None`) working with `singledispatchmethod`.